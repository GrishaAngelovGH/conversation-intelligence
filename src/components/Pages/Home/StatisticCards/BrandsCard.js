import {
  Chart as ChartJS,
  LinearScale,
  PointElement,
  Tooltip,
  Legend,
} from 'chart.js'

import { Bubble } from 'react-chartjs-2'

import StatisticCard from './StatisticCard'
ChartJS.register(LinearScale, PointElement, Tooltip, Legend)

export const options = {
  scales: {
    y: {
      beginAtZero: true
    }
  }
}

export const data = {
  datasets: [{
    label: 'Brand 1',
    data: Array.from({ length: 50 }, () => ({
      x: Math.random() * 10,
      y: Math.random() * 20,
      r: Math.random() * 30
    })),
    backgroundColor: 'rgba(255, 103, 20, 0.5)'
  },
  {
    label: 'Brand 2',
    data: Array.from({ length: 50 }, () => ({
      x: Math.random() * 10,
      y: Math.random() * 20,
      r: Math.random() * 30
    })),
    backgroundColor: 'rgba(53, 242, 135, 0.5)'
  }]
}

const BrandsCard = () => {
  const chart = (<Bubble options={options} data={data} />)

  const description = (
    <div>
      <h4>Customer mentions of the brands <span className="fw-bold">Brand 1</span> and <span className="fw-bold">Brand 2</span> have been detected across sales calls</h4>
      <p className="text-muted">When customers mention brands, it can surface new competitors and opportunities to update sales strategies</p>
    </div>
  )

  return (<StatisticCard chart={chart} description={description} />)
}

export default BrandsCard