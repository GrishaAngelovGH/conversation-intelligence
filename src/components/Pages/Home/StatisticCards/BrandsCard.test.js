import { render } from '@testing-library/react'

import BrandsCard from './BrandsCard'

test('should render BrandsCard component', () => {
  const view = render(<BrandsCard />)

  expect(view).toMatchSnapshot()
})