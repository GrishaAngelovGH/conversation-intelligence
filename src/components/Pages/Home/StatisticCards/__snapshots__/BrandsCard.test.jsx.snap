// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render BrandsCard component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row align-items-center mt-3 border border-2 rounded shadow"
      >
        <div
          class="col-md-6"
        >
          <canvas
            height="0"
            role="img"
            style="display: block; box-sizing: border-box; height: 0px; width: 0px;"
            width="0"
          />
        </div>
        <div
          class="col-md-6"
        >
          <div>
            <h4>
              Customer mentions of the brands 
              <span
                class="fw-bold"
              >
                Brand 1
              </span>
               and 
              <span
                class="fw-bold"
              >
                Brand 2
              </span>
               have been detected across sales calls
            </h4>
            <p
              class="text-muted"
            >
              When customers mention brands, it can surface new competitors and opportunities to update sales strategies
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row align-items-center mt-3 border border-2 rounded shadow"
    >
      <div
        class="col-md-6"
      >
        <canvas
          height="0"
          role="img"
          style="display: block; box-sizing: border-box; height: 0px; width: 0px;"
          width="0"
        />
      </div>
      <div
        class="col-md-6"
      >
        <div>
          <h4>
            Customer mentions of the brands 
            <span
              class="fw-bold"
            >
              Brand 1
            </span>
             and 
            <span
              class="fw-bold"
            >
              Brand 2
            </span>
             have been detected across sales calls
          </h4>
          <p
            class="text-muted"
          >
            When customers mention brands, it can surface new competitors and opportunities to update sales strategies
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
